link/src/main/java/com/stripe/android/link/injection/LinkCommonModule.kt

```diff
@@ -17,4 +26,21 @@ internal interface LinkCommonModule {
     @Binds
     @Singleton
     fun bindLinkEventsReporter(linkEventsReporter: DefaultLinkEventsReporter): LinkEventsReporter
+
+    companion object {
+        @Provides
+        @Singleton
+        fun provideConsumersApiService(
```

Comment:
Should we provide a shared `@Module` that can be referred from a `Component` from different SDKs? like [CoroutineContextModule](https://github.com/stripe/stripe-android/blob/3037a7a917b7d1ec36ee8495a306f667f6dfd147/stripe-core/src/main/java/com/stripe/android/core/injection/CoroutineContextModule.kt#L12)

Reply:
Done - https://github.com/stripe/stripe-android/pull/6089/commits/199624fc79ba905f594f2894ba3ed773e132dec0

Note that I moved the `Stripe.AppInfo` global field to `stripe-core` so that any SDK can provide values to it, not just payments.

Reply:
Hmm can we make each user of the module provide AppInfo in their graph upfront? that way you can declare the provides method like this  without needing to change `ConsumersApiServiceImpl` API
```
@Provides
        @Singleton
        fun provideConsumersApiService(
           appInfo: AppInfo,
           ...
        ): ConsumersApiService = ConsumersApiServiceImpl(
            appInfo = appInfo,
            ...
        )
```

Reply:
We could do that, but on `payments` AppInfo can be set dynamically at any time. If we provide the value on graph creation, it can get outdated. 

We could inject a provider function too (`appInfoProvider: () -> AppInfo`), but might not be the cleanest pattern (CC @jaynewstrom-stripe interested in your thoughts here too)

Reply:
Refactoring this into a shared object seems like a good idea for the SDK next major version. Before that, I'm guessing we'll struggle with half ideal work arounds.

No strong opinion here.

Reply:
Agreed, that's how I'm seeing it as well. I'll work on that API review : )